File: cardshowviewtakenpicturesview/src/main/java/br/com/stant/libraries/cardshowviewtakenpicturesview/domain/model/SaveOnlyMode.java
package br.com.stant.libraries.cardshowviewtakenpicturesview.domain.model;

import android.graphics.Bitmap;
import android.os.Parcel;
import android.os.Parcelable;

public class SaveOnlyMode implements Parcelable {

    private Bitmap enabledIcon;
    private String enabledWarning;
    private Bitmap disabledIcon;
    private String disabledWarning;

    public SaveOnlyMode(Bitmap enabledIcon, String enabledWarning, Bitmap disabledIcon, String disabledWarning) {
        this.enabledIcon     = enabledIcon;
        this.enabledWarning  = enabledWarning;
        this.disabledIcon    = disabledIcon;
        this.disabledWarning = disabledWarning;
    }

    public Bitmap getEnabledIcon() {
        return enabledIcon;
    }

    public String getEnabledWarning() {
        return enabledWarning;
    }

    public Bitmap getDisabledIcon() {
        return disabledIcon;
    }

    public String getDisabledWarning() {
        return disabledWarning;
    }

    protected SaveOnlyMode(Parcel in) {
        enabledIcon     = in.readParcelable(Bitmap.class.getClassLoader());
        enabledWarning  = in.readString();
        disabledIcon    = in.readParcelable(Bitmap.class.getClassLoader());
        disabledWarning = in.readString();
    }

    public static final Creator<SaveOnlyMode> CREATOR = new Creator<SaveOnlyMode>() {
        @Override
        public SaveOnlyMode createFromParcel(Parcel in) {
            return new SaveOnlyMode(in);
        }

        @Override
        public SaveOnlyMode[] newArray(int size) {
            return new SaveOnlyMode[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeParcelable(enabledIcon, i);
        parcel.writeString(enabledWarning);
        parcel.writeParcelable(disabledIcon, i);
        parcel.writeString(disabledWarning);
    }


}

File: cardshowviewtakenpicturesview/src/main/java/br/com/stant/libraries/cardshowviewtakenpicturesview/domain/model/SaveOnlyMode.kt
package br.com.stant.libraries.cardshowviewtakenpicturesview.domain.model

import android.graphics.Bitmap
import android.os.Parcel
import android.os.Parcelable

class SaveOnlyMode(val enabledIcon: Bitmap?, val enabledWarning: String?, val disabledIcon: Bitmap?, val disabledWarning: String?) : Parcelable {

    constructor(parcel: Parcel) : this(
            parcel.readParcelable(Bitmap::class.java.classLoader),
            parcel.readString(),
            parcel.readParcelable(Bitmap::class.java.classLoader),
            parcel.readString())

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeParcelable(enabledIcon, flags)
        parcel.writeString(enabledWarning)
        parcel.writeParcelable(disabledIcon, flags)
        parcel.writeString(disabledWarning)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<SaveOnlyMode> {
        override fun createFromParcel(parcel: Parcel): SaveOnlyMode {
            return SaveOnlyMode(parcel)
        }

        override fun newArray(size: Int): Array<SaveOnlyMode?> {
            return arrayOfNulls(size)
        }
    }


}